internal/expr/scan.go

```diff
@@ -148,28 +172,59 @@ func (re *ResultExpr) decodeValue(v reflect.Value) error {
 	return nil
 }
 
-func setValue(dest reflect.Value, fInfo field, val any) error {
+func setValue(dest reflect.Value, fInfo fielder, val any) error {
 	var isZero bool
 
 	v := reflect.ValueOf(val)
+	name := fInfo.Name()
 
-	if val == nil {
-		if fInfo.omitEmpty {
-			return nil
+	switch dest.Type().Kind() {
```

Comment:
This switch could maybe be over the type of `fInfo`, i.e. `switch f := fInfo.(type)` instead of the structs/map. We could either count on the caller to make sure that the dest was the right kind or do a check in each of the cases.